### PR TITLE
specification: Define the interface unbinding flows

### LINKED ISSUE #57: Add section for TDI Interface removal

### PR DESCRIPTION
**Pull request by sameo**
Fixes #57

### CODE REVIEW

File: specification/07-theory_operations.adoc [hunk 1]
```diff
@@ -880,6 +879,104 @@ TVM ->> TVM: Use device interface
 
 === Interface Unbinding
 
+As the platform resources owner, only the host supervisor domain manager may
+decide to unbind an interface from a TVM. At any moment, it can choose to do so
+through the `sbi_covh_unbind_interface()` `COVH` ABI. The targeted TSM services
+the request by moving the TDI from the TDISP `CONFIG_LOCKED` or `RUN` state back
```
Comment (jyao1) [inline specification/07-theory_operations.adoc:885]: I think it handles `ERROR` state as well.

File: specification/07-theory_operations.adoc [hunk 2]
```diff
@@ -880,6 +879,104 @@ TVM ->> TVM: Use device interface
...
+autonumber
+
+participant TDI as Device Interface
+participant DSM as Device DSM
+participant VMM as Host Supervisor Domain Manager (VMM)
+participant TSM
+participant TVM
+participant IOMMU
+
+note over TDI: CONFIG_LOCKED or RUN
```
Comment (jyao1) [inline specification/07-theory_operations.adoc:944]: or ERROR.

File: specification/07-theory_operations.adoc [hunk 3]
```diff
@@ -880,6 +879,104 @@ TVM ->> TVM: Use device interface
...
+
+opt
+VMM ->> TSM: [COVH] sbi_covh_run_vcpu()
+TVM ->> TSM: [COVH] sbi_covg_stop_interface()
+end
+
+VMM ->> TSM: [COVH] sbi_covh_unbind_interface()
+TSM ->> TSM: Generate TDISP STOP_INTERFACE_REQUEST
+TSM ->> VMM: [COVH] spdm_req(STOP_INTERFACE_REQUEST)
+VMM ->> DSM: [DOE] SPDM_STOP_INTERFACE_REQUEST
```
Comment (jyao1) [inline specification/07-theory_operations.adoc:954]: I feel the name `SPDM_STOP_INTERFACE_REQUEST` is confusing.

It is really a TDIPS protocol in SPDM session. But it is not SPDM protocol.
Maybe just call it `spdm_req(STOP_INTERFACE_REQUEST)`

File: specification/07-theory_operations.adoc (hunk 1 above)
Comment (sameo) [reply specification/07-theory_operations.adoc]: Fixed.

File: specification/07-theory_operations.adoc (hunk 2 above)
Comment (sameo) [reply specification/07-theory_operations.adoc]: Fixed, thanks.

File: specification/07-theory_operations.adoc (hunk 3 above)
Comment (sameo) [reply specification/07-theory_operations.adoc]: Makes sense, this is fixed now.

File: specification/07-theory_operations.adoc [hunk 4]
```diff
@@ -880,6 +879,104 @@ TVM ->> TVM: Use device interface
...
+DSM ->> TDI: STOP
+TDI ->> TDI: Abort or complete all pending DMA requests
+TDI ->> TDI: Scrub all TDI secrets and confidential data
+note over TDI: CONFIG_UNLOCKED
+DSM ->> VMM: [DOE] SPDM_STOP_INTERFACE_RESPONSE
```
[the remark this answers is not in the document]
Comment (sameo) [reply specification/07-theory_operations.adoc]: Fixed as well.

**Review by jyao1 — APPROVED** [review]

File: src/07-theory_operations.adoc [hunk 5]
```diff
@@ -880,6 +879,104 @@ TVM ->> TVM: Use device interface
...
+   upon xref:tdi-acceptation[acceptation of the TDI by TVM].
+7. Eventually, the host supervisor domain manager reclaims the TDI MMIO regions
+   exposed to the TVM address space by calling the
+   `sbi_covh_reclaim_tvm_interface_region()` `COVH` ABI for each and every one of
+   them. The TSM rejects any one of those requests if the reclaimed MMIO region
+   belongs to a bound interface, or if it does not belong to the MMIO gpa -> hpa
+   mapping the TSM initially created for the interface at
+   xref:binding-flow[TDI binding time].
+8. The TSM removes all C-IOMMU, TVM G-stage mappings and the MMIO gpa -> hpa
+   mapping for the interface.
```
Comment (rsahita) [inline src/07-theory_operations.adoc:927]: sorry for the late comment - does step 8 include removing all cached C-IOMMU ATC mappings for the interface being unbound?

**Review by rsahita — COMMENTED** [review]
lgtm with a minor comment/clarification

**Pull request merged**